src/evergreen/resource_permissions.py

```diff
@@ -0,0 +1,47 @@
+# -*- encoding: utf-8 -*-
+"""Resource permissions representation of evergreen."""
+from __future__ import absolute_import
+
+from typing import TYPE_CHECKING, Any, Dict, List
+
+from evergreen.base import _BaseEvergreenObject, evg_attrib
+
+if TYPE_CHECKING:
+    from evergreen.api import EvergreenApi
+
+
+class UserResourcePermissions(_BaseEvergreenObject):
+    """Representation of a single user's resource permissions."""
+
+    user = evg_attrib("user")
+    permissions = evg_attrib("permissions")
+
+    def __init__(self, json: Dict[str, Any], api: "EvergreenApi") -> None:
+        """
+        Create a UserResourcePermissions instance.
+
+        :param json: json representing project.
+        :param api: evergreen api object.
+        """
+        super().__init__(json, api)
+
+
+class AllUserResourcePermissions(_BaseEvergreenObject):
+    """Representation of all permissions held by users to a resource."""
+
+    def __init__(self, json: Dict[str, Any], api: "EvergreenApi") -> None:
+        """
+        Create an AllUserResourcePemrissions instance.
+
+        :param json: json representing project.
+        :param api: evergreen api object.
+        """
+        super().__init__(json, api)
+
+    @property
+    def permissions(self) -> List[UserResourcePermissions]:
```

Comment:
Would we want to keep this as a dictionary? It seems like it could be useful to quick access permission for a user.